lib/spack/spack/spec.py

```diff
@@ -5364,6 +5364,26 @@ def save_dependency_specfiles(
             fd.write(dep_spec.to_json(hash=ht.dag_hash))
 
 
+def partition_spec_list(input_specs, predicate_fn):
```

Comment:
Can you move this elsewhere? It's just a generic stable partition

Reply:
Moved to `llnl.util.lang` and renamed input args + reworked doctoring